src/tools/utils.js

```diff
@@ -88,15 +97,17 @@ function processChangedObjectFields(handler, desiredAssetState, currentAssetStat
         for (const currentObjectFieldPropertyName of Object.keys(
           currentAssetState[fieldName]
         )) {
+          // Loop through each object property that exists currently
           if (desiredAssetState[fieldName][currentObjectFieldPropertyName] === undefined) {
+            // If the object has a property that exists now but doesn't exist in the proposed state
             if (allowDelete) {
               desiredAssetStateWithChanges[fieldName][
                 currentObjectFieldPropertyName
               ] = null;
             } else {
+              // If deletes aren't allowed, do outright delete the property within the object
               log.warn(
-                `Detected that the ${fieldName} of the following ${
-                  handler.name
+                `Detected that the ${fieldName} of the following ${handler.name || handler.id || ''
```

Comment:
Added the ID, although it might not be super helpful in some cases, but it does prevent this from printing out `undefined` in the logs.